ts/Grid/Core/Pagination/Pagination.ts

```diff
@@ -125,21 +177,56 @@ class Pagination {
      */
     constructor(
         grid: Grid,
-        options: PaginationOptions
+        options: PaginationOptions & Pagination.PaginationState
     ) {
         this.grid = grid;
-        this.options = options ?? Pagination.defaultOptions;
-        this.itemsPerPage = this.options.itemsPerPage ||
-            Pagination.defaultOptions.itemsPerPage as number;
-    }
+        this.options = merge(Pagination.defaultOptions, options);
+
+        this.pageSizeOptions =
+            this.options.controls.pageSizeSelector.options as number[];
```

Comment:
No need to cast it here, since the `this.options.controls.pageSizeSelector.options` is either `number[]` or `undefined`. Can we check if it's defined or change the `pageSizeOptions` to be conditional?